python/lsst/resources/packageresource.py

```diff
@@ -39,6 +42,31 @@ class PackageResourcePath(ResourcePath):
     resource name.
     """
 
+    @classmethod
+    def _fixDirectorySep(
+        cls, parsed: urllib.parse.ParseResult, forceDirectory: bool = False
+    ) -> Tuple[urllib.parse.ParseResult, bool]:
+        """Ensure that a path separator is present on directory paths."""
+        parsed, dirLike = super()._fixDirectorySep(parsed, forceDirectory=forceDirectory)
+        if not dirLike:
+            try:
+                # If the resource does location does not exist this can
+                # fail immediately. It is possible we are doing path
+                # manipulation and not wanting to read the resource now,
+                # so catch the error and move on.
+                ref = resources.files(parsed.netloc).joinpath(parsed.path.lstrip("/"))
+            except ModuleNotFoundError:
+                pass
+            else:
+                dirLike = ref.is_dir()
+        return parsed, dirLike
+
+    def isdir(self) -> bool:
+        """Return True if this URI is a directory, else False."""
+        ref = resources.files(self.netloc).joinpath(self.relativeToPathRoot)
+        # The item might not exist yet.
```

Comment:
The comment in the method above this one suggests that it'll raise `ModuleNotFoundError`, right?  I guess I'm not sure what this comment is saying.

Reply:
Yes, when I wrote this code I had no idea that `ModuleNotFoundError` was a possibility...

Reply:
I've refactored to do that check in one place and added a test.